src/file/file.service.ts

```diff
@@ -46,30 +46,29 @@ export class FileService {
 
     let id: string | ObjectID = new ObjectID();
 
-    // Create the file id by reversing key.
-    if (key) {
-      id = this.reverseString(key);
-    }
-
     // If there is no parent given, create the file in the user's root folder.
     const parentID: string = folderID;
 
-    const file: IFile = new fileModel({
-      bucket,
-      key,
+    // basicFile is without key and bucket - in case it is a folder.
+    let basicFile: IFile = {
       type,
       name,
       ownerID,
       size,
-      _id: id,
-      parent: parentID,
-    });
+      parent: parentID
+    };
+
+    // Create the file id by reversing key, and add ket and bucket.
+    if (key) {
+      id = this.reverseString(key);
```

Comment:
Create a function `generateKey()/generateS3Key()` or whatever you find suitable and use it here.

Reply:
There's no need for the comment if you have a function for it, and even more if the function has it's own comment.

Reply:
done